### PR TITLE
docs(tutorial): add deprecation notice to v1.4.x tutorial and point to the correct branch

### PR DESCRIPTION
**Pull request by gkalpak**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Docs update.

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] ~~Tests for the changes have been added (for bug fixes / features)~~
- [x] Docs have been added / updated (for bug fixes / features)

**Other information**:
- Add a deprecation notice on the 'index' page.
- Hide the "Live Demo" buttons (since we don't have a live demo).
- Update the GitHub diff links to point to the ~~`pre-v1.5.0-snapshot`~~ `1.4-snapshot` angular-phonecat branch.
- Modify all git commangs to use the appropriate branch and tags.

Related to #14416.

### CODE REVIEW

**Comment by gkalpak** [comment]
The only thing missing (except for merging this PR) is to create the `old-step-*` tags on angular-phonecat.

(We might need better names for `pre-v1.5.0-snapshot` and `old-step-*`.)

**Comment by gkalpak** [comment]
@petebacondarwin, I have assigned you, since you are the official tutorial reviewer this month :stuck_out_tongue:

File: docs/app/src/tutorials.js [hunk 1]
```diff
@@ -11,16 +11,16 @@ angular.module('tutorials', [])
...
       '<a ng-href="tutorial/{{next}}"><li class="btn btn-primary">Next <i class="glyphicon glyphicon-step-forward"></i></li></a>',
     link: function(scope, element, attrs) {
       var seq = 1 * attrs.docTutorialNav;
       scope.seq = seq;
       scope.prev = pages[seq];
       scope.next = pages[2 + seq];
-      scope.diffLo = seq ? (seq - 1): '0~1';
-      scope.diffHi = seq;
+      scope.diffLo = 13 - seq;
+      scope.diffHi = 12 - seq;
```
Comment (petebacondarwin) [inline docs/app/src/tutorials.js]: why these changes?

File: docs/app/src/tutorials.js [hunk 2]
```diff
@@ -11,16 +11,16 @@ angular.module('tutorials', [])
     scope: {},
     template:
       '<a ng-href="tutorial/{{prev}}"><li class="btn btn-primary"><i class="glyphicon glyphicon-step-backward"></i> Previous</li></a>\n' +
-      '<a ng-href="http://angular.github.io/angular-phonecat/step-{{seq}}/app"><li class="btn btn-primary"><i class="glyphicon glyphicon-play"></i> Live Demo</li></a>\n' +
-      '<a ng-href="https://github.com/angular/angular-phonecat/compare/step-{{diffLo}}...step-{{diffHi}}"><li class="btn btn-primary"><i class="glyphicon glyphicon-search"></i> Code Diff</li></a>\n' +
+      // '<a ng-href="http://angular.github.io/angular-phonecat/step-{{seq}}/app"><li class="btn btn-primary"><i class="glyphicon glyphicon-play"></i> Live Demo</li></a>\n' +
```
Comment (petebacondarwin) [inline docs/app/src/tutorials.js]: leaving this commented for the future?

File: docs/content/tutorial/index.ngdoc [hunk 3]
```diff
@@ -5,6 +5,22 @@
...
+    It is recommended to use the latest tutorial version for the following reasons:
+    <ul>
+      <li>It showcases features introduced in AngularJS v1.5+</li>
+      <li>It follows modern best practices in terms of architecture and code organization</li>
+      <li>It has more up-to-date dependencies and tools</li>
+      <li>It is actively maintained</li>
+    </ul>
+  </p>
+</div>
+
```
Comment (petebacondarwin) [inline docs/content/tutorial/index.ngdoc]: LG

File: docs/content/tutorial/index.ngdoc [hunk 4]
```diff
@@ -73,7 +89,7 @@ Clone the [angular-phonecat repository][angular-phonecat] located at GitHub by r
 command:
 
 ```
-git clone --depth=14 https://github.com/angular/angular-phonecat.git
+git clone --depth=14 --branch=pre-v1.5.0-snapshot https://github.com/angular/angular-phonecat.git
```
Comment (petebacondarwin) [inline docs/content/tutorial/index.ngdoc]: LG

File: docs/content/tutorial/index.ngdoc [hunk 5]
```diff
@@ -82,6 +98,11 @@ This command creates the `angular-phonecat` directory in your current directory.
 download much smaller and faster.
 </div>
 
+<div class="alert alert-info">
+  The `--branch=pre-v1.5.0-snapshot` option tells Git to pull down the `pre-v1.5.0-snapshot` branch.
+  The code for this older version of the tutorial is on that branch.
+</div>
+
```
Comment (petebacondarwin) [inline docs/content/tutorial/index.ngdoc]: LG

File: docs/content/tutorial/step_00.ngdoc [hunk 6]
```diff
@@ -16,7 +16,7 @@ dependencies, as described in {@link index#get-started Get Started}.
 In the `angular-phonecat` directory, run this command:
 
 ```
-git checkout -f step-0
+git checkout -f old-step-0
```
Comment (petebacondarwin) [inline docs/content/tutorial/step_00.ngdoc]: LG

File: docs/app/src/tutorials.js (hunk 1 above)
Comment (gkalpak) [inline docs/app/src/tutorials.js]: Hm...I made them before I created the new tags. Now that we have the tags, I should just change `compare/step-` to `compare/old-step-` :smiley:

File: docs/app/src/tutorials.js (hunk 2 above)
Comment (gkalpak) [inline docs/app/src/tutorials.js]: For historical reasons :stuck_out_tongue: I will remove it.
Are we OK with not having a live demo for the old tutorial ?

Comment (petebacondarwin) [inline docs/app/src/tutorials.js]: Yes

**Comment by gkalpak** [comment]
@petebacondarwin, I made the changes. PTAL

**Comment by gkalpak** [comment]
Renamed branch to `1.4-snapshot` and tags to `1.4-step-*` (as discussed off-github).
Merging...

**Comment by gkalpak** [comment]
Closed via 6234cda.
